### PR TITLE
Enhanced logging

### PR DESCRIPTION
**Pull request by keirog**
- adds news logs
- converts existing logs to use res.locals.logger in a structured format

 🐿 v2.8.0

### CODE REVIEW

File: app/controllers/users.server.controller.js [hunk 1]
```diff
@@ -232,28 +259,39 @@ exports.search = (req, res) => {
 
 	User.count(options, (countErr, count) => {
 		const t2 = Date.now();
-		logger.debug('Users counted', { time: t2 - t1 });
+
+		logger('verbose', {
+			event: 'USERS COUNTED',
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we underscore this please

Alternatively, can we remove this as I am not sure what value it gives us in the logs

File: app/controllers/users.server.controller.js [hunk 2]
```diff
@@ -232,28 +259,39 @@ exports.search = (req, res) => {
...
-						message: findErr
+						message: findErr.message
 					});
 				}
 				else {
 					const t3 = Date.now();
-					logger.debug('Users found', { time: t3 - t2 });
+
+					logger('verbose', {
+						event: 'USERS FOUND',
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Same as above comment

File: app/controllers/users.server.controller.js [hunk 3]
```diff
@@ -266,14 +304,22 @@ exports.search = (req, res) => {
...
-									message: encryptErr
+									message: encryptErr.message
 								});
 							}
 							else {
 								const t4 = Date.now();
-								logger.debug('Users decrypted', { time: t4 - t3 });
+
+								logger('verbose', {
+									event: 'USERS DECRYPTED',
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Same same

File: app/controllers/users.server.controller.js (hunk 1 above)
Comment (keirog) [reply app/controllers/users.server.controller.js]: Oops, thanks! Would be good to keep it for a bit just to see if the performance info is useful—I have no idea how long these calls take

File: app/controllers/users.server.controller.js [hunk 4]
```diff
@@ -156,20 +174,28 @@ exports.list = (req, res) => {
...
 						// Callback
-						(encryptErr, decryptedUsers) => {
+						(decryptionErr, decryptedUsers) => {
 							/* istanbul ignore if */
-							if (encryptErr) {
-								logger.warn(encryptErr);
+							if (decryptionErr) {
+								logger('error', {
+									event: 'ERROR_DECRYPTING_USERS',
+									err: decryptionErr.message
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/users.server.controller.js [hunk 5]
```diff
@@ -232,28 +259,39 @@ exports.search = (req, res) => {
...
 			.limit(perPage)
 			.skip(perPage * page)
 			.lean()
 			.exec((findErr, users) => {
 				/* istanbul ignore if */
 				if (findErr) {
-					logger.warn(findErr);
+					logger('error', {
+						event: 'SEARCH_ERROR',
+						err: findErr.message
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/users.server.controller.js [hunk 6]
```diff
@@ -266,14 +304,22 @@ exports.search = (req, res) => {
 						(encryptErr, decryptedUsers) => {
 							/* istanbul ignore if */
 							if (encryptErr) {
-								logger.warn(encryptErr);
+								logger('error', {
+									event: 'DECRYPTION_ERROR',
+									err: encryptErr.message
+								});
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/users.server.controller.js [hunk 7]
```diff
@@ -317,10 +372,14 @@ exports.patch = (req, res) => {
 	user.save((updateErr) => {
 		/* istanbul ignore if */
 		if (updateErr) {
-			logger.warn(updateErr);
+			logger('error', {
+				event: 'SAVE_ERROR',
+				err:  updateErr.message
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/users.server.controller.js [hunk 8]
```diff
@@ -351,14 +421,23 @@ exports.updateOne = (req, res, next) => {
...
 			.send({ message: 'Please search by uuid or email' });
 	}
 
 	User.findOne(searchObj).exec((findErr, user) => {
 		/* istanbul ignore next */
 		if (findErr) {
-			logger.warn(findErr);
+			logger('error', {
+				event: 'ERROR_FINDING_USER',
+				err: findErr.message
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/users.server.controller.js [hunk 9]
```diff
@@ -390,17 +473,20 @@ exports.updateOne = (req, res, next) => {
...
 exports.userByUuid = (req, res, next, uuid) => {
+	const { logger } = res.locals;
 
 	User.findOne({ uuid: uuid }, { __v: 0, createdOn: 0 }, (findErr, user) => {
 		/* istanbul ignore next */
 		if (findErr) {
-			logger.warn(findErr);
+			logger('error', {
+				event: 'ERROR_FINDING_BY_UUID',
+				error: findErr.message
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/users.server.controller.js [hunk 10]
```diff
@@ -419,14 +507,19 @@ exports.userByUuid = (req, res, next, uuid) => {
...
 		/* istanbul ignore next */
 		if (err) {
+			logger('error', {
+				event: 'DELETE_UUID_NOT_FOUND',
+				message: err.message
+			});
+
 			return res.status(500).json({
-				error: err
+				error: err.message
```
Comment (AbdullahAli) [inline app/controllers/users.server.controller.js]: Can we change this key to `message` to keep it consistent please

File: app/controllers/usersByList.server.controller.js [hunk 11]
```diff
@@ -89,15 +92,17 @@ exports.list = (req, res) => {
...
 							if (encryptErr) {
+								logger('error', { event: 'USERS_BY_LIST_DECRYPTION_FAILED', time: t4 - t3 });
 								return res.status(400).json({
 									message: encryptErr
 								});
 							}
 							else {
-								const t4 = Date.now();
-								logger.debug('Users decrypted', { time: t4 - t3 });
+								logger('debug', { event: 'USERS_BY_LIST_DECRYPTION_DONE', time: t4 - t3 });
```
Comment (AbdullahAli) [inline app/controllers/usersByList.server.controller.js]: Should `debug` be error here?

Comment (keirog) [reply app/controllers/usersByList.server.controller.js]: I don’t understand this one… it’s not an error 🤔

Comment (keirog) [reply app/controllers/usersByList.server.controller.js]: Ah I see the confusion—the else is redundant here. I’ll remove it ✌️ But I’m intrigued to see what the timings are like before removing the logs. And hopefully 90% of the nesting 😓

Comment (AbdullahAli) [reply app/controllers/usersByList.server.controller.js]: 🙌

**Pull request merged**